sdk/core/azure-core/CMakeLists.txt

```diff
@@ -14,13 +14,20 @@ if(NOT CURL_FOUND)
   find_package(CURL ${CURL_MIN_REQUIRED_VERSION} REQUIRED)
 endif()
 
+if(BUILD_TRANSPORT_CURL)
+  SET(CURL_TRANSPORT_ADAPTER_SRC src/http/curl/curl.cpp)
+endif()
+if(BUILD_TRANSPORT_WINHTTP)
+  SET(WIN_TRANSPORT_ADAPTER_SRC src/http/winhttp/win_http_transport.cpp)
+endif()
+
 add_library (
   ${TARGET_NAME}
   src/context.cpp
   src/credentials/credentials.cpp
   src/credentials/policy/policies.cpp
   src/http/body_stream.cpp
-  src/http/curl/curl.cpp
+  ${CURL_TRANSPORT_ADAPTER_SRC}
```

Comment:
If someone doesn't define either `BUILD_TRANSPORT_CURL` or `BUILD_TRANSPORT_WINHTTP`, how do we make sure the r ight ones are included by "default"?

Reply:
that's what we use the CMake module for:
https://github.com/Azure/azure-sdk-for-cpp/pull/706/files?file-filters%5B%5D=.cmake&file-filters%5B%5D=.hpp&file-filters%5B%5D=.txt&file-filters%5B%5D=.yml#diff-6376d284abe67ef40b0c8fb9bbbc5edcR5

It handles the case when no option was added to set the default.
It basically turns ON WinHTTP on Windows, or Curl on Lin/Mac if user didn't set anything

Reply:
Right, gotcha.